amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/model/rma/type/RecordCategoryType.java

```diff
@@ -204,6 +196,49 @@ public Void doWork()
 
     }
 
+    /**
+     * Record Category move behaviour
+     *
+     * @see org.alfresco.repo.node.NodeServicePolicies.OnMoveNodePolicy#onMoveNode(org.alfresco.service.cmr.repository.ChildAssociationRef, org.alfresco.service.cmr.repository.ChildAssociationRef)
+     */
+    @Override
+    @Behaviour
+            (
+                    kind = BehaviourKind.CLASS,
+                    notificationFrequency = NotificationFrequency.FIRST_EVENT
+            )
+    public void onMoveNode(ChildAssociationRef oldChildAssocRef, ChildAssociationRef newChildAssocRef)
+    {
+        if (nodeService.getType(newChildAssocRef.getChildRef()).equals(TYPE_RECORD_CATEGORY) && dispositionService.getDispositionSchedule(oldChildAssocRef.getParentRef()) != null)
+        {
+            reinitializeRecordFolders(newChildAssocRef);
+        }
+    }
+
+    /**
+     *  Recursively reinitialize each folder in a structure of categories
+     *  Unwanted aspects will be removed from the child records and the records will be re-filed
+     *  Disposition schedule aspects and properties will be inherited from the new parent category
+     *
+     * @param childAssociationRef
+     */
+    private void reinitializeRecordFolders(ChildAssociationRef childAssociationRef)
+    {
+        for (ChildAssociationRef newChildRef : nodeService.getChildAssocs(childAssociationRef.getChildRef(),
+                ContentModel.ASSOC_CONTAINS,
+                RegexQNamePattern.MATCH_ALL))
+        {
+            if (nodeService.getType(newChildRef.getChildRef()).equals(TYPE_RECORD_CATEGORY))
```

Comment:
another way to check if records category/folder/record is to use the methods from ServiceBaseImpl  ```isRecordCategory``` as they are using the internalNodeService and were considered more performant (not sure if they can be called here without adding a new dependency)